### PR TITLE
Refactor Cascades debugger

### PR DESCRIPTION
**Pull request by hatyo**

### CODE REVIEW

File: fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/debug/PlannerRepl.java [hunk 1]
```diff
@@ -80,7 +80,9 @@ public class PlannerRepl implements Debugger {
             "88.  .88 88 88.  .88 88    88 88    88 88.  ... 88      \n" +
             "88Y888P' dP `88888P8 dP    dP dP    dP `88888P' dP      \n" +
             "88                                                      \n" +
-            "dP                                                      \n";
+            "dP                                                      \n" +
```
Comment (normen662) [inline fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/debug/PlannerRepl.java:83]: Want to give the whole thing a new banner? `planner` is so bland :-)

File: fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/debug/PlannerRepl.java [hunk 2]
```diff
@@ -156,7 +169,9 @@ public void onRegisterQuantifier(@Nonnull final Quantifier quantifier) {
     @Override
     public void onInstall() {
         try {
-            terminal = TerminalBuilder.builder().build();
+            if (terminal == null) {
```
Comment (normen662) [inline fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/debug/PlannerRepl.java:172]: Why not always insist on passing in the Terminal? Can we invert this and make the `terminal` `NonNull`?

File: fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/debug/PlannerRepl.java [hunk 3]
```diff
@@ -272,6 +287,9 @@ public void onEvent(final Event event) {
                         final Commands.Command<Event> command = commandOptional.get();
                         final Optional<Boolean> isContinueOptional = getSilently("run command", () -> command.executeCommand(this, event, parsedLine));
                         isContinue = isContinueOptional.orElse(false);
+                        if (command instanceof Commands.QuitCommand) {
```
Comment (normen662) [inline fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/debug/PlannerRepl.java:290]: why not just communicate `isContinue` back to here and then let the real do its thing?

**Review by normen662 — CHANGES_REQUESTED** [review]

File: fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/debug/PlannerRepl.java (hunk 1 above)
Comment (hatyo) [reply fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/debug/PlannerRepl.java:83]: I kinda like It honestly 😄 , but please let me know if you have something better, something like this?

```
╭━━━╮╱╱╱╱╱╱╱╱╱╱╱╱╱╭╮╱╱╱╱╱╱╭━━━╮╱╱╭╮
┃╭━╮┃╱╱╱╱╱╱╱╱╱╱╱╱╱┃┃╱╱╱╱╱╱╰╮╭╮┃╱╱┃┃
┃┃╱╰╋━━┳━━┳━━┳━━┳━╯┣━━┳━━╮╱┃┃┃┣━━┫╰━┳╮╭┳━━┳━━┳━━┳━╮
┃┃╱╭┫╭╮┃━━┫╭━┫╭╮┃╭╮┃┃━┫━━┫╱┃┃┃┃┃━┫╭╮┃┃┃┃╭╮┃╭╮┃┃━┫╭╯
┃╰━╯┃╭╮┣━━┃╰━┫╭╮┃╰╯┃┃━╋━━┃╭╯╰╯┃┃━┫╰╯┃╰╯┃╰╯┃╰╯┃┃━┫┃
╰━━━┻╯╰┻━━┻━━┻╯╰┻━━┻━━┻━━╯╰━━━┻━━┻━━┻━━┻━╮┣━╮┣━━┻╯
╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╭━╯┣━╯┃
```

File: fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/debug/PlannerRepl.java (hunk 2 above)
Comment (hatyo) [reply fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/debug/PlannerRepl.java:172]: Sure, I can also do that.

File: fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/debug/PlannerRepl.java (hunk 3 above)
Comment (hatyo) [reply fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/debug/PlannerRepl.java:290]: That's even better, I will try that.

Comment (hatyo) [reply fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/debug/PlannerRepl.java:290]: hmm... it looks like `isContinueOptional` just indicates whether the planner should continue or prompt for the next command, i.e. it does not break the debugger main loop:

```java
/**
* ...
* @return {@code true} if planning should continue afterwards, {@code false} if the REPL should prompt
*         for the next command after the execution of this command has finished.
*/
boolean executeCommand(@Nonnull PlannerRepl plannerRepl, @Nonnull E event, @Nonnull ParsedLine parsedLine);
```

File: fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/debug/PlannerRepl.java (hunk 1 above)
Comment (normen662) [reply fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/debug/PlannerRepl.java:81]: party time -- excellent

Comment (normen662) [reply fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/debug/PlannerRepl.java:81]: not for now -- it's ok.

File: fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/debug/PlannerRepl.java (hunk 3 above)
Comment (normen662) [reply fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/debug/PlannerRepl.java:281]: That's all it ever does. There is no really breaking out other than `System.exit()`. You can throw an exception as an alternative.

**Review by normen662 — APPROVED** [review]

**Pull request merged**